### PR TITLE
[JBIDE-21703] Retrieve istags associated with a resource

### PR DESCRIPTION
**Pull request by jcantrill**
@adietish can you test to see if this resolves the issue.  I'm not certain what the actions are to verify.

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java [hunk 1]
```diff
@@ -382,16 +393,16 @@ public String toString() {
 			}
 			return client.get(resource.getKind(), resource.getName(), resource.getNamespace());
 		} catch (UnauthorizedException e) {
-			return retryGet("Unauthorized.  Trying to reauthenticate", e, resource);
+			return retryGet("Unauthorized.  Trying to reauthenticate", e, resource.getKind(), resource.getName(), resource.getNamespace());
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java]: trying to understand: how is this related to the inability to retrieve istags? Did the backend change?

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/IOpenShiftConnection.java [hunk 2]
```diff
@@ -31,6 +31,15 @@
...
 	<T extends IResource> List<T> getResources(String kind, String namespace);
+
+	/**
+	 * Retrieve a resource by name
+	 * @param kind
+	 * @param namespace
+	 * @param name
+	 * @return
+	 */
+	<T extends IResource> T getResource(String kind, String namespace, String name);
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/IOpenShiftConnection.java]: ok, so we now need this method?

**Comment by adietish** [comment]
great, that was quick, thanks!
+1

**Comment by adietish** [comment]
unfortunately this still isnt working for me, an empty set of istags is retrieved. Looking into it.

**Comment by fbricon** [comment]
Works for me, the deployment directory is properly set in the server wizard and the server editor in PR #944

**Comment by adietish** [comment]
I intentified the problem: if you get back to the connection page and choose a different one, the deployment resource mappers are not repopulated correctly, causing this to fail in this particular case.

**Comment by adietish** [comment]
merged to jbosstools-4.3.x and master.
filing a new issue for the lacking reloading of the mappers in case you change the connection
